C#/C#.sublime-syntax

```diff
@@ -1377,6 +1346,49 @@ contexts:
     - match: (?=\}|\)|>|\]|,|{{reserved}})
       pop: true
 
+  tuple:
+    - meta_scope: meta.group.tuple.cs
+    - match: \)
+      scope: punctuation.section.group.end.cs
+      pop: true
+    - match: ','
+      scope: punctuation.separator.tuple.cs
+    - match: ':'
+      scope: keyword.operator.assignment.cs
+    - match: _\b
+      scope: variable.language.deconstruction.discard.cs
+    - match: (?!{{reserved}})(?={{namespaced_name}}{{type_suffix}}\s+{{name}}\s*[:,])
+      push: var_declaration_explicit
+    - match: '({{name}})(<)'
+      captures:
+        1: support.type.cs
+        2: meta.generic.cs punctuation.definition.generic.begin.cs
+      push:
```

Comment:
We probably should try to introduce "named contexts for everything". Not sure if C# is a candidate, but it helps with inheriting syntaxes as well as debugging via scope popup, which now shows context names.

Reply:
I like this point. May be worth adding to https://github.com/sublimehq/Packages/issues/757

Reply:
I wonder if we should create a more structured kind of documentation for things like that in the long term.

I have too much such details in mind but somehow find it hard to create documentation from it.

Reply:
Refactoring syntaxes to use named contexts can be done in dedicated PRs as well.